### PR TITLE
HTCONDOR-2467 cred cleanup

### PR DESCRIPTION
**Pull request by JaimeFrey**
<Insert PR description here, and leave checklist below for code review.>

# HTCondor Pull Request Checklist for internal reviewers

- [x] Verify that (GitHub thinks) the merge is clean. If it isn't, and you're confident you can resolve the conflicts, do so. Otherwise, send it back to the original developer.
- [x] Verify that the related Jira ticket exists and has a target version number and that it is correct.
- [x] Verify that the Jira ticket is in review status and is assigned to the reviewer.
- [x] Verify that the Jira ticket (HTCONDOR-xxx) is mentioned at the beginning of the title. Edit it, if not
- [x] Verify that the branch destination of the PR matches the target version of the ticket
- [x] Check for correctness of change
- [ ] Check for regression test(s) of new features and bugfixes (if the feature doesn't require root)
- [x] Check for documentation, if needed  (documentation [build logs](https://readthedocs.org/projects/htcondor/builds/))
- [x] Check for version history, if needed
- [x] Check BaTLab dashboard for successful build (https://batlab.chtc.wisc.edu/results/workspace.php) and test for either the PR or a workspace build by the developer that has the Jira ticket as a comment.
- [x] Check that each commit message references the Jira ticket (HTCONDOR-xxx)

## After the above
- Hit the merge button if the pull request is approved and it is not a security patch (security changes require 2 additional reviews)
- If the pull request is approved, take the ticket out of review state
- Assign JIRA Ticket back to the developer

### CODE REVIEW

File: src/condor_schedd.V6/schedd.cpp [hunk 1]
```diff
@@ -1667,12 +1667,10 @@ Scheduler::count_jobs()
 		} else if ( current_time - owner_info.LastHitTime > AbsentOwnerLifetime ) {
 			// mark user creds for sweeping.
 			if (cred_dir_krb) {
-				dprintf(D_FULLDEBUG, "creating credmon KRB mark file for user %s\n", owner_info.Name());
-				credmon_mark_creds_for_sweeping(cred_dir_krb, owner_info.Name());
+				credmon_mark_creds_for_sweeping(cred_dir_krb, owner_info.Name(), credmon_type_KRB);
 			}
 			if (cred_dir_oauth) {
-				dprintf(D_FULLDEBUG, "creating credmon OAUTH mark file for user %s\n", owner_info.Name());
```
Comment (jasoncpatton) [inline src/condor_schedd.V6/schedd.cpp:1674]: Can there still be some logging made available for when the `count_jobs` timer thinks a mark file should be written (either here or in `credmon_mark_creds_for_sweeping`)?

**Review by jasoncpatton — CHANGES_REQUESTED** [review]
On a job submit transaction, mark files are cleared.

Mark files are only created if the user's credential directory (if "OAuth" style credentials) or krb credentials exist.

Unless I'm missing somewhere where this already happens, I think it would be good to log when the `count_jobs` timer thinks credentials should be marked for cleanup.

File: src/condor_schedd.V6/schedd.cpp (hunk 1 above)
Comment (JaimeFrey) [reply src/condor_schedd.V6/schedd.cpp:1674]: I didn't want to log the message in the case where we decide there's nothing to clean up. But I can add the message inside credmon_mark_creds_for_sweeping() after that check.

Comment (jasoncpatton) [reply src/condor_schedd.V6/schedd.cpp:1674]: Thanks!

**Review by jasoncpatton — APPROVED** [review]
LGTM... I like the extra code removal, too!

**Pull request merged**